spec/policy/oidc_authentication/oidc_authentication_spec.lua

```diff
@@ -0,0 +1,185 @@
+local _M = require('apicast.policy.oidc_authentication')
+local JWT = require('resty.jwt')
+local rsa = require('fixtures.rsa')
+local oidc_discovery = require('resty.oidc.discovery')
+local OIDC = require('apicast.oauth.oidc')
+local test_backend_client = require 'resty.http_ng.backend.test'
+
+local access_token = setmetatable({
+  header = { typ = 'JWT', alg = 'RS256', kid = 'somekid' },
+  payload = {
+    iss = 'http://example.com/issuer',
+    sub = 'some',
+    aud = 'one',
+    exp = ngx.now() + 3600,
+  },
+}, { __tostring = function(jwt) return JWT:sign(rsa.private, jwt) end })
+
+describe('oidc_authentication policy', function()
+  describe('.new', function()
+    it('works without configuration', function()
```

Comment:
For these 2 cases with empty or nil config, it might be useful to check that `rewrite` and `access` do not raise unexpected errors.